File: c++/2864.cpp
// Link: https://leetcode.com/problems/maximum-odd-binary-number/description/
// Runtime: 2 ms
// Memory: 8.63 MB

class Solution 
{
public:
    std::string maximumOddBinaryNumber(const std::string& s) 
    {
        int numberOfOnes = 0;
        for (const auto& symbol : s)
            if (symbol == '1')
                numberOfOnes++;
        
        return std::string(numberOfOnes - 1, '1') + std::string(s.size() - numberOfOnes, '0') + '1';
    }
};

File: python/2864.py
# Link: https://leetcode.com/problems/maximum-odd-binary-number/description/
# Runtime: 26 ms
# Memory: 16.56 MB


class Solution:
    def maximumOddBinaryNumber(self, s: str) -> str:
        number_of_ones = s.count("1")

        return "1" * (number_of_ones - 1) + "0" * (len(s) - number_of_ones) + "1"
